spot_wrapper/wrapper.py

```diff
@@ -289,71 +537,157 @@ def __init__(self, username, password, hostname, robot_name, logger, start_estop
+            self._point_cloud_requests.append(build_pc_request(source))
+
+        self._hand_image_requests = []
+        for source in hand_image_sources:
+            self._hand_image_requests.append(
+                build_image_request(source, image_format=image_pb2.Image.FORMAT_RAW)
+            )
 
         self._camera_image_requests = []
         for camera_source in CAMERA_IMAGE_SOURCES:
-            self._camera_image_requests.append(build_image_request(
-                camera_source,
-                image_format=image_pb2.Image.FORMAT_JPEG,
-                pixel_format=image_pb2.Image.PIXEL_FORMAT_RGB_U8,
-                quality_percent=50,
-            ))
+            self._camera_image_requests.append(
+                build_image_request(
+                    camera_source,
+                    image_format=image_pb2.Image.FORMAT_JPEG,
+                    pixel_format=image_pb2.Image.PIXEL_FORMAT_RGB_U8,
+                    quality_percent=50,
+                )
+            )
 
         self._depth_image_requests = []
         for camera_source in DEPTH_IMAGE_SOURCES:
-            self._depth_image_requests.append(build_image_request(
-                camera_source, pixel_format=image_pb2.Image.PIXEL_FORMAT_DEPTH_U16
-            ))
+            self._depth_image_requests.append(
+                build_image_request(
+                    camera_source, pixel_format=image_pb2.Image.PIXEL_FORMAT_DEPTH_U16
+                )
+            )
 
         self._depth_registered_image_requests = []
         for camera_source in DEPTH_REGISTERED_IMAGE_SOURCES:
-            self._depth_registered_image_requests.append(build_image_request(
-                camera_source, pixel_format=image_pb2.Image.PIXEL_FORMAT_DEPTH_U16
-            ))
+            self._depth_registered_image_requests.append(
+                build_image_request(
+                    camera_source, pixel_format=image_pb2.Image.PIXEL_FORMAT_DEPTH_U16
+                )
+            )
 
         try:
-            self._sdk = create_standard_sdk('ros_spot')
+            self._sdk = create_standard_sdk("ros_spot")
         except Exception as e:
             self._logger.error("Error creating SDK object: %s", e)
             self._valid = False
             return
 
+        self._logger.info("Initialising robot at {}".format(self._hostname))
         self._robot = self._sdk.create_robot(self._hostname)
 
-        try:
-            self._robot.authenticate(self._username, self._password)
-            self._robot.start_time_sync()
-        except RpcError as err:
-            self._logger.error("Failed to communicate with robot: %s", err)
-            self._valid = False
-            return
-
-        self._logger.info("Establishing time sync with robot - this could take some time.")
-        self._robot.time_sync.wait_for_sync(10.0)
-
-        if self._robot:
-            # Clients
+        authenticated = False
+        while not authenticated:
```

Comment:
I don't think it should, `time.sleep` can receive keyboard interrupts. I haven't specifically tested this but I've been using the driver with this code for a while and it's not been an issue. To test this the driver has to run before the robot is fully booted up. 

Part of the reason I added this is because the previous implementation meant that if you started the driver before all services and so on were available, it would just crash or maybe even fail silently rather than doing the smart thing and waiting until everything was available.